discord: add config for initial status/presence

ashen-dawn opened the pull request:
So we're not entirely happy with needing to mock out a message, but we also didn't want to duplicate the the existing code for managing presence sync and activity matching.  We're open to other ways to do it

ashen-dawn committed:
discord: add config for initial status/presence

Qyriad commented:
We'd definitely prefer doing away with the mock message. We can pull out the activity parsing into its own method, but I think the presence handling should probably just be duplicated — plus then `on_ready` can also try to fetch the initial presence instead of waiting for a presence change too.

Only other thing than that is style — multiline function calls should have the closing parenthesis on its own line (with the same indentation as the initial line). We won't have `mockMessage` anymore so that doesn't apply, but for future reference Python variables by convention are in `lower_snake_case`.

Thank you for the PR! 🙂

Qyriad commented:
Just pushed 6b57746c739cebc0b5dbe7336d121e7beb34c808 which pulls the activity parsing out into its own method — you should be able to rebase off of that

ashen-dawn committed:
discord: remove mock message class

ashen-dawn commented:
Okay I have updated it to remove the mock message, changed the formatting for the `__init__()` function declaration, and I *think* have not introduced any more camel case variables.

Qyriad committed:
fix minor style things
